shared/src/sources/balancer/graph_api.rs

```diff
@@ -87,6 +88,88 @@ impl BalancerSubgraphClient {
         })
     }
 
+    /// Retrieves the list of registered stable pools from the subgraph.
+    pub async fn get_stable_pools(&self) -> Result<RegisteredStablePools> {
+        let block_number = self.get_safe_block().await?;
+        let mut pools = vec![];
+
+        // We do paging by last ID instead of using `skip`. This is the
+        // suggested approach to paging best performance:
+        // <https://thegraph.com/docs/graphql-api#pagination>
+        let mut last_id = H256::default();
+        while {
+            let page = self
+                .0
+                .query::<pools_query::Data>(
+                    pools_query::STABLE_QUERY,
+                    Some(json_map! {
+                        "block" => block_number,
+                        "pageSize" => QUERY_PAGE_SIZE,
+                        "lastId" => json!(last_id),
+                    }),
+                )
+                .await?
+                .pools;
+
+            let has_next_page = page.len() == QUERY_PAGE_SIZE;
+            if let Some(last_pool) = page.last() {
+                last_id = last_pool.id;
+            }
+
+            for pool in page {
+                pools.push(pool.into_registered_stable(block_number)?);
+            }
+
+            has_next_page
+        } {}
+
+        Ok(RegisteredStablePools {
+            fetched_block_number: block_number,
+            pools,
+        })
+    }
+
+    /// Retrieves the list of registered stable pools from the subgraph.
+    pub async fn get_registered_pools(&self) -> Result<RegisteredPools> {
```

Comment:
Here, I have provided two possible avenues of approach. Taking this would fetch all pools and allow us to filter after the fact. Although, since we have multiple pool initializers, this would result in fetching all pools for each pool type and excluding those we don't want. However, it does keep the code here less "duplicated".